webapp/components/Map/MapBaseLayerPeriodSelector.js

```diff
@@ -50,44 +100,74 @@ export const MapBaseLayerPeriodSelector = () => {
         ready: true,
         periods: availablePeriods,
         periodByValue: availablePeriods.reduce((acc, period) => ({ ...acc, [getPeriodValue(period)]: period }), {}),
-        selectedPeriodValue: lastPeriodValue,
+        selectedPeriodValueLeft: lastPeriodValue,
+        selectedPeriodValueRight: lastPeriodValue,
+        procLeft: Proc.rgb,
+        procRight: Proc.rgb,
       })
     })()
-  }, [periodSelectorAvailable, provider, periodType])
+  }, [periodSelectorAvailable, provider, periodType, ready])
 
   const onMapLayerPeriodChange = useCallback(
-    (event) => {
+    (isLeft) => (event) => {
       const periodValue = event.target.value
       // replace url in current layer
-      const layer = Object.values(map._layers).find((layer) => layer._url && layer.id === contextBaseLayer.id)
+      let layer
+      if (isLeft) {
+        layer = Object.values(map._layers).find((layer) => layer.options.id !== 'side')
+      } else {
+        layer = Object.values(map._layers).find((layer) => layer.options.id === 'side')
+      }
       if (layer) {
         const period = periodByValue[periodValue]
-        const url = baseLayerUrlByProviderFunction[provider]({ period, apiKey })
+        let url = baseLayerUrlByProviderFunction[provider]({ period, apiKey })
+        url += '&proc=' + (isLeft ? procLeft : procRight)
         layer.setUrl(url)
         onBaseLayerUpdate({ ...contextBaseLayer, url })
         // update state
-        setState((statePrev) => ({ ...statePrev, selectedPeriodValue: Number(periodValue) }))
+        if (isLeft) {
+          setState((statePrev) => ({ ...statePrev, selectedPeriodValueLeft: Number(periodValue) }))
+          sideBySideObject.setLeftLayers(layer)
+        } else {
+          setState((statePrev) => ({ ...statePrev, selectedPeriodValueRight: Number(periodValue) }))
+          sideBySideObject.setRightLayers(layer)
+        }
       }
     },
-    [map, contextBaseLayer, provider, onBaseLayerUpdate, setState]
+    [map, contextBaseLayer, provider, onBaseLayerUpdate, state]
   )
 
-  // disable map dragging when dragging period slider
-  const onSliderMouseDown = useCallback(() => {
-    map.dragging.disable()
-  }, [map])
-
-  const onSliderMouseUp = useCallback(() => {
-    map.dragging.enable()
-  }, [map])
+  const onCheckboxValueChange = (isLeft) => () => {
+    let newProc
+    if (isLeft) {
+      newProc = !leftChecked ? Proc.cir : Proc.rgb
+      setState((statePev) => ({ ...statePev, leftChecked: !statePev.leftChecked, procLeft: newProc }))
+    } else {
+      newProc = !rightChecked ? Proc.cir : Proc.rgb
+      setState((statePev) => ({ ...statePev, rightChecked: !statePev.rightChecked, procRight: newProc }))
+    }
+    let layer
+    if (isLeft) {
+      layer = Object.values(map._layers).find((layer) => layer.options.id !== 'side')
+    } else {
+      layer = Object.values(map._layers).find((layer) => layer.options.id === 'side')
```

Comment:
this way of getting the layer is repeated above, can you extract it into a function?